src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/CertificateHelper.cs

```diff
@@ -0,0 +1,305 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Actors.KVSToRCMigration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Cryptography;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+
+    /// <summary>
+    /// Helper methods related to certificate usage (loading, verifying, etc)
+    /// </summary>
+    internal class CertificateHelper
+    {
+        private static readonly string TraceType = typeof(CertificateHelper).Name;
+
+        internal delegate bool IsCertificateAMatchForFindValue(X509Certificate2 enumeratedCert, string findValue);
+
+        internal static List<X509Certificate2> GetCertificates(MigrationSecuritySettings securitySettings)
+        {
+            var storeName = securitySettings.CertificateStoreName;
+            var storeLocation = securitySettings.CertificateStoreLocation;
+            var findType = securitySettings.CertificateFindType;
+            var findValue = securitySettings.CertificateFindValue;
+            bool includeExpiredCerts = false;
+
+            return FindMatchingCertificates(storeName, storeLocation, findType, findValue, includeExpiredCerts);
+        }
+
+        internal static List<X509Certificate2> GetClientCertificates(MigrationSecuritySettings securitySettings)
+        {
+            var storeName = securitySettings.CertificateStoreName;
+            var storeLocation = securitySettings.CertificateStoreLocation;
+            bool includeExpiredCerts = false;
+            var clientCertList = new List<X509Certificate2>();
+
+            foreach (var val in securitySettings.CertificateRemoteThumbprints.Split(','))
+            {
+                clientCertList.AddRange(FindMatchingCertificates(storeName, storeLocation, X509FindType.FindByThumbprint, val, includeExpiredCerts));
+            }
+
+            foreach (var val in securitySettings.CertificateRemoteCommonNames.Split(','))
+            {
+                clientCertList.AddRange(FindMatchingCertificates(storeName, storeLocation, X509FindType.FindBySubjectName, val, includeExpiredCerts));
+            }
+
+            return clientCertList;
+        }
+
+        internal static List<X509Certificate2> FindMatchingCertificates(
+            string storeName,
```

Comment:
The logic in this file is equivalent to certificate manager in system.fabric.common namespace?

Reply:
yes